app/routes/articles/components/ArticleEditor.tsx

```diff
@@ -129,15 +126,17 @@ const ArticleEditor = () => {
       pinned: data.pinned,
     };
 
-    dispatch(isNew ? createArticle(body) : editArticle(body)).then((res) => {
+    dispatch(
+      isNew ? createArticle(body) : editArticle({ id: articleId, ...body }),
+    ).then((res) => {
       navigate(
         isNew ? `/articles/${res.payload.result}/` : `/articles/${articleId}`,
       );
     });
   };
 
   const handleDeleteArticle = async () => {
-    await dispatch(deleteArticle(articleId)).then(() => {
+    await dispatch(deleteArticle(articleId!)).then(() => {
```

Comment:
```suggestion
    articleId && await dispatch(deleteArticle(articleId)).then(() => {
```

Reply:
No, I want it to throw an error if you somehow manage to press that button with `articleId = undefined`

Reply:
I feel like we should find some better way to handle this though. `articleId` comes from `useParams` and is actually guaranteed to be set as you cannot get to this page without an article id in the url. This is a problem in quite a lot of pages

Reply:
How do you feel about a typecast when getting the params?
```typescript
type ArticleEditorParams = { articleId: string };
const ArticleEditor = () => {
  const { articleId } = useParams<ArticleEditorParams>() as ArticleEditorParams;
  ...
```
which would then allow us to just get rid of the "!"